langstream-api-gateway/src/main/java/ai/langstream/apigateway/gateways/TopicProducerCache.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.langstream.apigateway.gateways;
+
+import ai.langstream.api.runner.topics.TopicProducer;
+import java.util.function.Supplier;
+
+public interface TopicProducerCache {
+    record Key(String tenant, String application, String gatewayId) {}
+
+    TopicProducer getOrCreate(Key key, Supplier<TopicProducer> topicProducerSupplier);
```

Comment:
this operation may fail, it is better to add a "throws" clause. The risk is to forget possible failures (that will happen for instance if the broker is down)